code/WorkInProgress/AnimationLibrary.dm

```diff
@@ -30,6 +30,41 @@
 	animate(A, color=list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1), time=time, easing=SINE_EASING)
 	return
 
+/proc/animate_fade_drug(var/atom/A, var/time=5) //This fades from normal colors to red being green, green being blue and blue being red
+	if (!istype(A) && !isclient(A))
+		return
+	A.color = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1)
+	animate(A, color=list(0,1,0,0, 0,0,1,0, 1,0,0,0, 0,0,0,1), time=time, easing=SINE_EASING)
+	return
```

Comment:
this one doesn't appear to be used?

Reply:
Oh yeah I figured that could be useful if somebody wanted to fade into the druggy vision without it being such a hard cut, didn't figure out/want to code that myself but making the animation was easy so I figured it could be good for someone in the future. Can cut that out if it's preferred.